app/models/response_node.rb

```diff
@@ -47,4 +59,33 @@ def _destroy
   def _destroy=(d)
     @destroy = d.is_a?(String) ? d == "true" : d
   end
+
+  def propogate_response_id
+    children.reject(&:destroyed?).each do |c|
+      c.response_id = response_id
+    end
+  end
+
+  def destroy_obsolete_children
+    children.destroy(children.select(&:irrelevant_or_marked_destroy?))
+  end
+
+  # TODO: remove. inst_num and this block will go away with answer_arranger
+  def update_inst_nums
+    children.reject(&:destroyed?).sort_by(&:new_rank).each_with_index do |c, i|
+      new_inst_num =
+        if c.parent.is_a?(AnswerGroupSet) # repeat group
+          i + 1
+        elsif %w[Answer AnswerSet AnswerGroupSet].include?(c.type)
+          c.parent.inst_num
+        else
+          1
+        end
+      c.inst_num = new_inst_num
+    end
+  end
+
+  def irrelevant_or_marked_destroy?
+    !relevant? || _destroy
```

Comment:
For some reason `destroy?` didn't work here. I need to move on to other project work so I'd prefer not to figure it out today, if that's ok?